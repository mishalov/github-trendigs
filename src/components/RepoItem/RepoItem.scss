@import "../../styles/variables.scss";

.repo-item {
  width: 100%;
  padding: 16px 36px;
  border-top: 1px solid $border-color;
  &__header {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }
  &__title {
    overflow: hidden;
    text-overflow: ellipsis;
    @include for-mobile() {
      a {
        display: block;
        padding: 4px 0px;
      }
    }
  }
  &__title-divider {
    padding: 0px 8px;
    @include for-mobile() {
      display: none;
    }
  }
  &__description {
    text-align: justify;
    line-height: 22px;
    padding-top: 8px;
    word-wrap: break-word;
    word-break: normal;
  }
  &__star {
    font-size: 12px;
    width: 100%;
    display: flex;
    align-items: center;
    img {
      margin-right: 8px;
    }
  }
}
