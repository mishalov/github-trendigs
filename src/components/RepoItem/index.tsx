import React from "react";
import { TRepo } from "github-trendings/types";
import star from "../../assets/svg/star_white.svg";
import "./RepoItem.scss";
import Button from "../Button";
import Tagline from "../Tagline";

interface IRepoItemProps {
  repoInfo: TRepo;
}

const RepoItem = (props: IRepoItemProps) => {
  const { repoInfo } = props;
  const {
    name,
    owner,
    htmlUrl,
    description,
    language,
    stargazersCount,
    forks
  } = repoInfo;
  const { login, avatarUrl } = owner;
  const ownerHtmlUrl = owner.htmlUrl;
  return (
    <article className="repo-item">
      <header className="repo-item__header">
        <h1 className="repo-item__title">
          <a href={ownerHtmlUrl} target="_blank" rel="noopener noreferrer">
            {login}
          </a>
          <span className="repo-item__title-divider">/</span>
          <a href={htmlUrl} target="_blank" rel="noopener noreferrer">
            {name}
          </a>
        </h1>
        <Button onClick={() => {}}>
          <div className="repo-item__star">
            <img alt="star" src={star} width="12" height="12" /> Star
          </div>
        </Button>
      </header>
      <p className="repo-item__description">{description}</p>
      <Tagline
        language={language}
        stargazersCount={stargazersCount}
        forks={forks}
      />
    </article>
  );
};

export default RepoItem;
