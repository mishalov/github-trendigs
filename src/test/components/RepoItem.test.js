import React from "react";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
import RepoItem from "../../components/RepoItem";

Enzyme.configure({ adapter: new Adapter() });
const wrappers = [];

const repoInfoMockup = {
  id: 194151280,
  name: "OnDemandMinecraft",
  fullName: "trevor-laher/OnDemandMinecraft",
  htmlUrl: "https://github.com/trevor-laher/OnDemandMinecraft",
  description:
    "An AWS hosted Minecraft server that will only run when players are active. Players can start the server through a simple UI accessed through free Heroku server hosting.",
  language: "Python",
  stargazersCount: 273,
  owner: { login: "trevor-laher", htmlUrl: "https://github.com/trevor-laher" },
  forks: 16,
  stargazersUrl:
    "https://api.github.com/repos/trevor-laher/OnDemandMinecraft/stargazers",
  forksUrl: "https://api.github.com/repos/trevor-laher/OnDemandMinecraft/forks"
};

beforeEach(() => {
  wrappers.push(
    shallow(
      <RepoItem
        repoInfo={repoInfoMockup}
        starred={false}
        starThis={() => {
          return "test_return";
        }}
      />
    )
  );
  wrappers.push(
    shallow(
      <RepoItem repoInfo={repoInfoMockup} starred={true} starThis={() => {}} />
    )
  );
});

describe("<RepoItem /> rendering", () => {
  it("renders correctly not starred", () => {
    expect(wrappers[0]).toMatchSnapshot();
    console.log('wrappers[0].find(".button"): ', wrappers[0].find(".button"));
    wrappers[0]
      .find("Button")
      .first()
      .simulate("click");
  });
  it("renders correctly starred", () => {
    expect(wrappers[1]).toMatchSnapshot();
  });
});
