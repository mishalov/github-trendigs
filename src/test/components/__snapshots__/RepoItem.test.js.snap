// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoItem /> rendering renders correctly not starred 1`] = `
<article
  className="repo-item"
>
  <header
    className="repo-item__header"
  >
    <h1
      className="repo-item__title"
    >
      <a
        href="https://github.com/trevor-laher"
        rel="noopener noreferrer"
        target="_blank"
      >
        trevor-laher
      </a>
      <span
        className="repo-item__title-divider"
      >
        /
      </span>
      <a
        href="https://github.com/trevor-laher/OnDemandMinecraft"
        rel="noopener noreferrer"
        target="_blank"
      >
        OnDemandMinecraft
      </a>
    </h1>
    <Button
      onClick={[Function]}
      status="normal"
    >
      <div
        className="repo-item__star"
      >
        <img
          alt="star"
          height="12"
          src="star_white.svg"
          width="12"
        />
        star
      </div>
    </Button>
  </header>
  <p
    className="repo-item__description"
  >
    An AWS hosted Minecraft server that will only run when players are active. Players can start the server through a simple UI accessed through free Heroku server hosting.
  </p>
  <Tagline
    forks={16}
    language="Python"
    stargazersCount={273}
  />
</article>
`;

exports[`<RepoItem /> rendering renders correctly starred 1`] = `
<article
  className="repo-item"
>
  <header
    className="repo-item__header"
  >
    <h1
      className="repo-item__title"
    >
      <a
        href="https://github.com/trevor-laher"
        rel="noopener noreferrer"
        target="_blank"
      >
        trevor-laher
      </a>
      <span
        className="repo-item__title-divider"
      >
        /
      </span>
      <a
        href="https://github.com/trevor-laher/OnDemandMinecraft"
        rel="noopener noreferrer"
        target="_blank"
      >
        OnDemandMinecraft
      </a>
    </h1>
    <Button
      onClick={[Function]}
      status="success"
    >
      <div
        className="repo-item__star"
      >
        <img
          alt="star"
          height="12"
          src="star_white.svg"
          width="12"
        />
        starred
      </div>
    </Button>
  </header>
  <p
    className="repo-item__description"
  >
    An AWS hosted Minecraft server that will only run when players are active. Players can start the server through a simple UI accessed through free Heroku server hosting.
  </p>
  <Tagline
    forks={16}
    language="Python"
    stargazersCount={273}
  />
</article>
`;
